src/HttpClientInterception/DelegateHelpers.cs

```diff
@@ -65,18 +66,58 @@ internal static class DelegateHelpers
         /// <returns>
         /// The converted delegate if <paramref name="onIntercepted"/> has a value; otherwise <see langword="null"/>.
         /// </returns>
-        internal static Func<HttpRequestMessage, Task<bool>>? ConvertToBooleanTask(Func<HttpRequestMessage, Task>? onIntercepted)
+        internal static Func<HttpRequestMessage, CancellationToken, Task<bool>>? ConvertToBooleanTask(Func<HttpRequestMessage, Task>? onIntercepted)
         {
             if (onIntercepted == null)
             {
                 return null;
             }
 
-            return async (message) =>
+            return async (message, _) =>
             {
                 await onIntercepted(message).ConfigureAwait(false);
                 return true;
             };
         }
+
```

Comment:
Having only seen the diffs in this file, rather than digging through all the code in an IDE, is there a simple way to refactor things to remove the need for two almost-identical methods for the `CancellationToken`s (or to have the existing ones call these new ones)?

Reply:
I'll take your preference, it could be we don't offer the equivalent overload that doesn't have the `CancellationToken`

Reply:
I guess it depends how many times they're used elsewhere in the code. Alternatively could (for the sake of a double null check) could just call the other method and reduce the duplication so it's something like:

```csharp
if (onIntercepted == null)
{
    return null;
}

return ConvertToBooleanTask((message, _) => onIntercepted(message));
```